### DISCUSSION TITLE
Speaker connected to the display via jack but no control with the program !

### DISCUSSION THREAD

**Discussion by ghost** [MonitorControl/MonitorControl#579] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues

### Discussion

Thank for your great job.
Sry for my bad English but I try to explain the situation, i have a Samsung display (U28E590) without integrated loudspeakers, so i connected a loudspeaker to the display via jack, the bars for the sound are displayed and moved up/down when i use volume keys on the keyboard but have no effect on the sound (No changed)
I don't know if this is an issue or something like this is not supported.
Thank you in advance for your answer.

**Comment by waydabber** [comment]
Hi @is4lloum - this seems to be a display specific issue. Some displays with external speakers connected might need a special speakerSelect (0x63) command to work and also check for audioJackConnectionStatus (0x65). This is rare however, most of the time this is not the case (I have an LG with a jack and no built in audio and it works for example) as displays tend to autodetect the audio output (since yours don't even have an internal speaker, it makes no sense anyway to use these commands, but still the display firmware might be buggy in this regard). You might want to try to use ddcctl and send direct commands to the display to figure out what needs to be done for proper control. If you come to a conclusion, we might add a feature request to add a special option for this in the advanced section of Displays.

I'll convert this issue to a Discussion. Please let me know if you come up with something. Thank you!

**Comment by ghost** [comment]
Hey @waydabber,
thank you for taking the time to answer me. You are right that there is a problem with the display because I installed ddcctl and could only change the brightness with it, but nothing changes with sound.
Is there any program where the problem can be narrowed down, as I don't have much experience with that.
I have tested eqMac and could use it to control the audio but MonitorController much better.
Thank you :)

**Reply by waydabber** [reply]
Unfortunately there is no specific app to help with this except using ddcctl (or alternatively you can use Lunar's command line interface - @alin23) and sending the above-mentioned hex command codes with varying values to see if there is a combination of them that makes a difference. :(

**Comment by ghost** [comment]
Brightness and contrast are working great with MonitorController only the sound does not work, although the window bar is shown but has no effect on the sound
I'll read more about it and try to find something.
again thank you for your time

**Comment by AHarmlessPyro** [comment]
It seems like this might be a wider Samsung wide thing. I have a C24F390 and seem to have a similar lack of control over the volume.

**Comment by SamThilmany** [comment]
It might be a Samsung-wide thing, but I can also report that the same issue is present on my Dell U2414H. The brightness control works fine, however, neither the volume nor the contrast is controlled by the sliders of MonitorControl.

**Reply by alin23** [reply]
Note that contrast might be blocked by a monitor setting:

<img width="894" alt="image" src="https://user-images.githubusercontent.com/3819725/150403815-59e76196-7773-4e87-908f-9f8831a34f2c.png">

I don't know of any setting that could block volume control, but I've had reports from users of Lunar that had success when turning the monitor speaker off then back on *(not sure how you would do that, I don't have a monitor with a built-in speaker)*. 

Most of them reported that the volume control started working but only for the built-in speaker, not for the AUX output.

**Reply by SamThilmany** [reply]
In my case, it is also a speaker connected via AUX, since the mentioned monitor does not have an integrated speaker.

From which application is this screenshot taken?

**Reply by alin23** [reply]
The screenshot is taken from the Lunar FAQ here: https://lunar.fyi/faq#brightness-not-changing
